internal/criocli/status.go

```diff
@@ -159,3 +175,38 @@ func goroutines(c *cli.Context) error {
 
 	return nil
 }
+
+func heap(c *cli.Context) error {
+	crioClient, err := crioClient(c)
+	if err != nil {
+		return err
+	}
+
+	data, err := crioClient.HeapInfo(c.Context)
+	if err != nil {
+		return err
+	}
+
+	outputPath := c.String("file")
+	if outputPath == "" {
+		outputPath = "heap.out"
+	}
+
+	var file *os.File
+
+	if outputPath == "-" {
+		file = os.Stdout
+	} else {
+		file, err = os.Create(outputPath)
+		if err != nil {
+			return fmt.Errorf("create output file %s: %w", outputPath, err)
+		}
+	}
+
+	if _, err := file.Write(data); err != nil {
+		return fmt.Errorf("write heap dump: %w", err)
+	}
+
+	logrus.Infof("Wrote heap dump to: %s", outputPath)
```

Comment:
Here, when we have "-" as the desired output destination, we need to make sure we don't print anything else, otherwise the heap dump format might be potentially invalid due to the extra output.

Reply:
Sounds good, changed as suggested.